### PR TITLE
Update markdown test tools

### PR DESCRIPTION
**Pull request by RDIL**
Signed-off-by: Reece Dunham <me@rdil.rocks>

# PR Summary

* Update the external markdown testing tools to the latest version

## PR Context

Staying somewhat up-to-date

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by xtqqczze** [comment]
```
2020-12-04T21:26:40.2433904Z       [-] https://github.com/powershell/powershell-rfc â†’ Status: 429 should work 60.18s
2020-12-04T21:26:40.2467358Z         RuntimeException: Failed to complete request to "https://github.com/powershell/powershell-rfc â†’ Status: 429". Response status code does not indicate success: 404 (Not Found).
2020-12-04T21:26:40.2469184Z         at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
```

**Comment by xtqqczze** [comment]
> ```
> 2020-12-04T21:26:40.2433904Z       [-] https://github.com/powershell/powershell-rfc â†’ Status: 429 should work 60.18s
> 2020-12-04T21:26:40.2467358Z         RuntimeException: Failed to complete request to "https://github.com/powershell/powershell-rfc â†’ Status: 429". Response status code does not indicate success: 404 (Not Found).
> 2020-12-04T21:26:40.2469184Z         at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 119
> ```
> 
> Related upstream issue: [tcort/markdown-link-check#101](https://github.com/tcort/markdown-link-check/issues/101)

The issue is fixed in tcort/markdown-link-check#106 but there is not yet a markdown-link-check release with these changes (see tcort/markdown-link-check#141)

**Comment by xtqqczze** [comment]
It seems markdown-link-check is making a HTTP GET request for every URL. This is not ideal behaviour for large binaries, to say the least:

https://github.com/PowerShell/PowerShell/blob/ea3036dc6d5bcbd58ee2b3b1960a3647467d0801/README.md#L65-L112

**Comment by xtqqczze** [comment]
@RDIL Please update markdown-link-check to v3.8.4.

**Comment by xtqqczze** [comment]
PowerShell-CI-static-analysis still has HTTP error 429 -> Too many requests

**Comment by iSazonov** [comment]
> PowerShell-CI-static-analysis still has HTTP error 429 -> Too many requests

We need update markdown-link-check config or commandline with 
> `retryOn429` if this is `true` then retry request when response is an HTTP code 429 after the duration indicated by `retry-after` header.

**Comment by xtqqczze** [comment]
Documentation should be updated as well:

https://github.com/PowerShell/PowerShell/blob/bb78d4b5c0aa9237df47c512d4059248078f5bbd/.github/CONTRIBUTING.md#L91-L102

**Comment by RDIL** [comment]
@PoshChan please rerun linux

**Comment by PoshChan** [comment]
@RDIL, you are not authorized to request a rebuild

**Comment by RDIL** [comment]
Fine, I'll do it myself.

File: docs/maintainers/README.md
Comment (iSazonov) [inline docs/maintainers/README.md:6]:
```suggestion
They have [write access](https://docs.github.com/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/managing-access-to-your-organizations-repositories) to the PowerShell repositories which gives them the power to:
```

File: docs/community/governance.md
Comment (iSazonov) [inline docs/community/governance.md:97]:
```suggestion
They have [write access](https://docs.github.com/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/managing-access-to-your-organizations-repositories) to the PowerShell repository which gives them the power to:
```

Comment (RDIL) [reply docs/community/governance.md:97]: I had it like this because it just redirects to the country of origin (or browser?)'s locale, and I had to do this because the redirect wasn't working, so this would just introduce another redirect.  Plus, the rest of the repository is in english, and there is a language selector on the GH docs page.

File: test/common/markdown/markdown-link.tests.ps1
Comment (xtqqczze) [inline test/common/markdown/markdown-link.tests.ps1:101]: Do we need 429 in the allowed failures, now we are using the `-r` option?

Comment (RDIL) [reply test/common/markdown/markdown-link.tests.ps1:101]: I'm going to experiment with this, it was just temporary to get things actually workng.

**Review by TravisEz13 — APPROVED** [review]

File: test/common/markdown/markdown-link.tests.ps1
Comment (RDIL) [reply test/common/markdown/markdown-link.tests.ps1]: Looks like it works

**Comment by xtqqczze** [comment]
**PowerShell-CI-static-analysis (Markdown and Common Tests):**

Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://www.computerperformance.co.uk/powershell/index.htm should work

**Comment by xtqqczze** [comment]
> **PowerShell-CI-static-analysis (Markdown and Common Tests):**
> 
> Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://www.computerperformance.co.uk/powershell/index.htm should work
> 
> Connection timed out

Look like a transient failure, but I've submitted #14431 to update the 301 redirect.

**Comment by RDIL** [comment]
> 
> 
> > **PowerShell-CI-static-analysis (Markdown and Common Tests):**
> > Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://www.computerperformance.co.uk/powershell/index.htm should work
> > Connection timed out
> 
> Look like a transient failure, but I've submitted #14431 to update the 301 redirect.
> 
> >  Computer Performance**[Windows PowerShell Tutorial - Introduction Microsoft's new scripting language](https://www.computerperformance.co.uk/powershell/index-13/)**Windows PowerShell Tutorial. An introducton to Microsoft's latest Windows scripting language. Learn PowerShell and master WMI. Replace VBScript and CMD with the PS command line.

I'll rebase once that gets merged.

**Comment by xtqqczze** [comment]
> I'll rebase once that gets merged.

@RDIL No need wait for that PR, failures in PowerShell-CI-static-analysis have not persisted.

**Comment by RDIL** [comment]
Whoops, that wasn't supposed to happen - sorry for anybody that got pinged due to my rebase, it usually doesn't do that!

File: docs/community/governance.md
Comment (xtqqczze) [reply docs/community/governance.md:97]: https://docs.github.com/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization is a redirect without the `en/free-pro-team@latest/` part.

File: docs/maintainers/README.md
Comment (xtqqczze) [reply docs/maintainers/README.md:6]:
```suggestion
They have [write access](https://docs.github.com/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization) to the PowerShell repositories which gives them the power to:
```

**Review by xtqqczze — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]

**Pull request merged**
